File: examples/typescript/pagination/snapshot.ts
import { Nightona } from '@nightona/sdk'

async function main() {
  const nightona = new Nightona()

  const result = await nightona.snapshot.list(2, 10)
  console.log(`Found ${result.total} snapshots`)
  result.items.forEach((snapshot) => console.log(`${snapshot.name} (${snapshot.imageName})`))
}

main().catch(console.error)

File: examples/python/pagination/_async/snapshot.py
import asyncio

from nightona import AsyncNightona


async def main():
    async with AsyncNightona() as nightona:
        result = await nightona.snapshot.list(page=2, limit=10)
        for snapshot in result.items:
            print(f"{snapshot.name} ({snapshot.image_name})")


if __name__ == "__main__":
    asyncio.run(main())
